nvflare/app_common/executors/script_executor.py

```diff
@@ -34,62 +29,51 @@
     from nvflare.app_opt.tf.params_converter import KerasModelToNumpyParamsConverter, NumpyToKerasModelParamsConverter
 
 
-class ScriptExecutor(InProcessClientAPIExecutor):
+class ScriptExecutor:
     def __init__(
         self,
-        task_script_path: str,
-        task_script_args: str = "",
-        task_wait_time: Optional[float] = None,
-        result_pull_interval: float = 0.5,
-        log_pull_interval: Optional[float] = None,
+        script: str,
+        script_args: str = "",
+        launch_external_process: bool = False,
         params_transfer_type: TransferType = TransferType.FULL,
-        from_nvflare_converter_id: Optional[str] = None,
-        to_nvflare_converter_id: Optional[str] = None,
-        train_with_evaluation: bool = True,
-        train_task_name: str = "train",
-        evaluate_task_name: str = "evaluate",
-        submit_model_task_name: str = "submit_model",
         params_exchange_format=DEFAULT_PARAMS_EXCHANGE_FORMAT,
     ):
-        """Wrapper around InProcessClientAPIExecutor for different params_exchange_format. Currently defaulting to `params_exchange_format=ExchangeFormat.PYTORCH`.
+        """ScriptExecutor is used with FedJob API to run or launch a script.
+
+        in-process `launch_external_process=False` uses InProcessClientAPIExecutor (default).
+        ex-process `launch_external_process=True` uses ClientAPILauncherExecutor.
 
         Args:
+            script (str): Script to run. For in-process must be a python script path. For ex-process can be any command supported by python subprocess.
+            script_args (str): Optional arguments for script.
+            launch_external_process (bool): Whether to launch the script in external process. Defaults to False.
+            params_transfer_type (str): How to transfer the parameters. FULL means the whole model parameters are sent.
+                DIFF means that only the difference is sent. Defaults to TransferType.FULL.
+            params_exchange_format (str): Format to exchange the parameters. Defaults based on detected imports.
```

Comment:
I personally prefer to use something more general like `ExecutorJobObj` to pass in either InProcessClientAPIExecutor or LauncherExecutor, if those names are not good, we just rename them, instead of keep wrapping into new classes?

And this exchange format could be useful in other type of executor as well.